gateway/ln-gateway/src/state_machine/complete.rs

```diff
@@ -199,56 +201,46 @@ impl CompleteHtlcState {
     async fn await_complete_htlc(
         context: GatewayClientContext,
         common: GatewayCompleteCommon,
-        outcome: HtlcOutcome,
+        htlc_outcome: HtlcOutcome,
     ) -> Result<(), CompleteHtlcError> {
-        // Wait until the lightning node is online to complete the HTLC
-        loop {
-            let htlc_outcome = outcome.clone();
-            let lightning_context = context.gateway.get_lightning_context().await;
-            match lightning_context {
-                Ok(lightning_context) => {
-                    let htlc = match htlc_outcome {
-                        HtlcOutcome::Success(preimage) => InterceptPaymentResponse {
-                            action: PaymentAction::Settle(preimage),
-                            payment_hash: common.payment_hash,
-                            incoming_chan_id: common.incoming_chan_id,
-                            htlc_id: common.htlc_id,
-                        },
-                        HtlcOutcome::Failure(_) => InterceptPaymentResponse {
-                            action: PaymentAction::Cancel,
-                            payment_hash: common.payment_hash,
-                            incoming_chan_id: common.incoming_chan_id,
-                            htlc_id: common.htlc_id,
-                        },
-                    };
-
-                    lightning_context
-                        .lnrpc
-                        .complete_htlc(htlc)
-                        .await
-                        .map_err(|_| CompleteHtlcError::FailedToCompleteHtlc)?;
-                    return Ok(());
-                }
+        // Wait until the lightning node is online to complete the HTLC.
+        let lightning_context = loop {
```

Comment:
I don't think these are the same semantics as the deleted code. Before, if `complete_htlc` failed we would try again, now it only loops when getting the lightning context

Reply:
We were looping only on lightning context previously, it was just a lot less clear by looking. We were using the question mark operator before, so `complete_htlc` actually was returning an error, not retrying.